### PR TITLE
Payments feature

### PR DESCRIPTION
**Pull request by tomohung**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -55,4 +56,27 @@ def set_following_relationship
...
+    # Get the credit card details submitted by the form
+    token = params[:stripeToken]
+
+    # Create the charge on Stripe's servers - this will charge the user's card
+    begin
+      charge = Stripe::Charge.create(
+        :amount => 3000, # amount in cents, again
+        :currency => "usd",
+        :source => token,
+        :description => 'Signs up for #{@user.full_name}'
```
Comment (brandoncc) [inline app/controllers/users_controller.rb]: This is not going to work like you hope. You need to use double quotes for interpolation to work.

File: app/views/users/new.html.haml [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+Stripe.setPublishableKey("#{ENV['STRIPE_PUBLISHABLE_KEY']}");
+</script>
+
+= javascript_include_tag 'payment'
+
 %section.register.container
   .row
     .col-sm-10.col-sm-offset-1
-      = bootstrap_form_for @user, html: {class: 'form-horizontal'} do |f|
+      = bootstrap_form_for @user, html: {class: 'form-horizontal', id: 'payment-form'} do |f|
```
Comment (brandoncc) [inline app/views/users/new.html.haml]: This is not the right way to do horizontal forms anymore. You need to use `layout: :horizontal`. https://github.com/bootstrap-ruby/rails-bootstrap-forms#horizontal-forms

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -79,30 +84,41 @@
...
+
+      before do
+        StripeMock.start
+        ActionMailer::Base.deliveries.clear
+      end
 
       it 'sends out email to the user with valid inputs' do
         user_attributes = Fabricate.attributes_for(:user)
-        post :create, user: user_attributes
+        post :create, user: user_attributes, stripToken: stripe_helper.generate_card_token
```
Comment (brandoncc) [inline spec/controllers/users_controller_spec.rb]: If these specs pass then you have a problem. You have a typo in all of them, `stripToken` should be `stripeToken`.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (tomohung) [inline app/controllers/users_controller.rb]: I make a big mistake here, thanks !!

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (tomohung) [inline spec/controllers/users_controller_spec.rb]: I remove it and still pass, I guess it's because I am using strip-ruby-mock. I'll check it out later.

**Pull request merged**